cristian-bars/Precurso/Pasapalabra/pasapalabra.js

```diff
@@ -0,0 +1,153 @@
+function runGame(){
+    rigthWords = 0;
+    wrongWords = 0;
+    pendent = 0;
+    
+    console.log("Bienvenido al juego del Pasabalabra!!");
+    
+    getName();
+    
+    runQuestions();
+    
+    
+    //Juego finalizado
+    console.log("Aquí finaliza el juego, el jugador " + yourName + " ha terminado la partida con un total de " + rigthWords + " palabras acertadas y un total de " + wrongWords + " palabras incorrectas!!");
+    
+    //Añadir jugador con sus resultados al array de jugadores
+    var obj = {};
+    obj["name"] = yourName;
+    obj["rigth"] = rigthWords;
+    obj["wrong"] = wrongWords;
+    
+    players.push(obj);
+    
+    //Preguntamos si quieren jugar otra partida
+    let playAgain = confirm("Quieres jugar otra partida??");
+        if (playAgain === true) {
+            for(let i = 0; i<questions.length;i++){
+                questions[i].status = 0;
+            }
+            runGame();
+        } else {
+            console.log("final");
+            sortParticipants(players);
+    
+            for(let i=0;i<players.length;i++){
+                console.log((i+1) + " " + players[i].name + " con " + players[i].rigth + " palabras acertadas y " + players[i].wrong + " palabras incorrectas");
+            }
+        } 
+    
+    
+}
+
+//Función para introducir el nombre del participante
+function getName(){
+    let yourName = prompt("Por favor, indicanos tu nombre");
```

Comment:
si la declaras dentro de la funcion, te funciona la variable yourName cuando la llamas fuera? he buscado esto en google: 

https://www.google.com/search?client=safari&rls=en&q=if+a+declare+a+var+inside+a+function,+can+i+call+it+outside%3F&ie=UTF-8&oe=UTF-8